habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDependenciesMojo.java

```diff
@@ -0,0 +1,156 @@
+        doExecute();
+    }
+
+    @Override
+    protected void doExecute() throws MojoExecutionException, MojoFailureException {
+        if (this.anchorSourceDirectory == null) {
+            this.anchorSourceDirectory = new File(session.getExecutionRootDirectory());
+        }
+
+        Set<MavenProject> habushuProjects = getHabushuProjects();
+        try {
+            copyRelevant(habushuProjects);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Copies the relevant source files by leveraging {@link FileSet}s to filter appropriately.
+     * @param habushuProjects corresponding projects of the pom's habushu-type dependencies
+     * @throws IOException
+     */
+    private void copyRelevant(Set<MavenProject> habushuProjects) throws IOException {
+        this.anchorOutputDirectory = new File(project.getBuild().getDirectory()
+                + "/venv-support/" + this.anchorSourceDirectory.getName());
+        Path srcRoot = this.anchorSourceDirectory.toPath();
+        Path destRoot = this.anchorOutputDirectory.toPath();
+
+        Map<Path, FileSet> dependencyFileSets = new HashMap<>();
+        for (MavenProject project : habushuProjects) {
+            Path projectPath = project.getFile().getParentFile().toPath();
+            Path relativeProjectPath = srcRoot.relativize(projectPath);
+            FileSet fileSet = getDefaultFileSet();
+            fileSet.setDirectory(projectPath.toString());
+            dependencyFileSets.put(relativeProjectPath, fileSet);
+        }
+
+        FileSetManager fileSetManager = new FileSetManager();
+        for (Path project : dependencyFileSets.keySet()) {
+            FileSet fileSet = dependencyFileSets.get(project);
+            logger.info("Copying {} files from {}",
+                    fileSetManager.getIncludedFiles(fileSet).length,
+                    project.getFileName()
+            );
+            for (String includedFile : fileSetManager.getIncludedFiles(fileSet)) {
+                Path relativePath = project.resolve(includedFile);
+                Files.createDirectories(destRoot.resolve(relativePath).getParent());
+                Files.copy(srcRoot.resolve(relativePath), destRoot.resolve(relativePath));
+            }
+        }
+    }
+
+    private FileSet getDefaultFileSet() {
+        FileSet fileSet = new FileSet();
+        fileSet.addExclude("target/**");
+        fileSet.addExclude(".venv/**");
+        fileSet.addExclude("dist/**");
+        return fileSet;
+    }
+
+    /**
+     * Checks listed habushu-type dependencies against the set of projects included in the Maven build's session
+     * @return the corresponding Maven projects that match the habushu-type dependencies
+     */
+    protected Set<MavenProject> getHabushuProjects() {
+        HashSet<MavenProject> habushuProjects = new HashSet<>();
+        collectHabushuDependenciesAsProjects(project, habushuProjects);
```

Comment:
Can drop the `project` parameter here and just reference it as `this.project` in the `collectHabushuDependenciesAsProjects` implementation.